pkg/util/filter/schema.go

```diff
@@ -17,39 +17,20 @@ package filter
 import (
 	"strings"
 
-	"github.com/pingcap/tidb/pkg/parser/mysql"
+	"github.com/pingcap/tidb/pkg/meta/metadef"
 )
 
 var (
 	// DMHeartbeatSchema is the heartbeat schema name
-	DMHeartbeatSchema = "DM_HEARTBEAT"
-	// InformationSchemaName is the `INFORMATION_SCHEMA` database name.
-	InformationSchemaName = "INFORMATION_SCHEMA"
-	// PerformanceSchemaName is the `PERFORMANCE_SCHEMA` database name.
-	PerformanceSchemaName = "PERFORMANCE_SCHEMA"
-	// MetricSchemaName is the `METRICS_SCHEMA` database name.
-	MetricSchemaName = "METRICS_SCHEMA"
+	DMHeartbeatSchema = "dm_heartbeat"
 	// InspectionSchemaName is the `INSPECTION_SCHEMA` database name
-	InspectionSchemaName = "INSPECTION_SCHEMA"
-	// WorkloadSchemaName is the `WORKLOAD_SCHEMA` database name
-	WorkloadSchemaName = strings.ToUpper(mysql.WorkloadSchema)
+	InspectionSchemaName = "inspection_schema"
 )
 
 // IsSystemSchema checks whether schema is system schema or not.
 // case insensitive
 func IsSystemSchema(schema string) bool {
-	schema = strings.ToUpper(schema)
-	switch schema {
-	case DMHeartbeatSchema, // do not create table in it manually
-		"SYS",   // https://dev.mysql.com/doc/refman/8.0/en/sys-schema.html
-		"MYSQL", // the name of system database.
-		WorkloadSchemaName,
-		InformationSchemaName,
-		InspectionSchemaName,
-		PerformanceSchemaName,
-		MetricSchemaName:
-		return true
-	default:
-		return false
-	}
+	schema = strings.ToLower(schema)
+	return schema == DMHeartbeatSchema || schema == InspectionSchemaName ||
+		metadef.IsMemOrSysDB(schema)
```

Comment:
Is this function only used for test? If so, maybe add a comment

Reply:
also found some places are using this API in tidb, but seems they should use `metadef.IsMemOrSysDB` directly, DM related table is not a system table to tidb
https://github.com/pingcap/tidb/blob/dfc00356e566406992664c3b438f1a410d7beeaf/pkg/planner/core/collect_column_stats_usage.go#L130
https://github.com/pingcap/tidb/blob/1b35bd01b4a9e444e007822ab8c7baf17f293944/pkg/ddl/cluster.go#L182